tests/integration/medusa_minio_integration_test.go

```diff
@@ -0,0 +1,61 @@
+package integration
+
+import (
+	. "github.com/k8ssandra/k8ssandra/tests/integration/steps"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var (
+	medusaMinioTestTable    = "medusa_test"
+	medusaMinioTestKeyspace = "medusa"
+)
+
+var _ = Describe("Medusa on MinIO", func() {
+	Context("Deploy K8ssandra with Medusa", func() {
+		It("Create a kind cluster and a namespace", func() {
+			AKindClusterIsRunningAndReachableStep("one worker")
+			IInstallTraefikStep()
+			ICreateTheNamespaceStep()
+			ICanSeeTheNamespaceInTheListOfNamespacesStep()
+		})
+		It("Create the Medusa secret ", func() {
+			IDeployMinIOUsingHelmAndCreateTheBucketStep("k8ssandra-medusa")
+			ICreateTheMedusaSecretInTheNamespaceApplyingTheFileStep("secret/medusa_minio_secret.yaml")
+			ICanSeeTheSecretInTheListOfSecretsInTheNamespaceStep("medusa-bucket-key")
+		})
+		It("Install K8ssandra with Medusa on MinIO", func() {
```

Comment:
Actually, I consider that such e2e/integration/functional tests should fail as soon as one step fails. It's, IMO, how BDD testing should be. One step expects the previous steps to have succeeded, otherwise we're not in the state we expect to be.
If I can't get the cassandradatacenter resource to be ready, then there's no point in trying to access Cassandra or run a repair.

What you describe is more in line with how unit testing should be done where you can reduce the feedback loop by being able to see all failing tests within a single run.

Reply:
Yes, but that's not what happens. The ``It`` parts are ran by ginkgo even if previous one fails.

Reply:
I actually use Ginkgo's failFast flag in CI (and when running the tests locally) to prevent subsequent tests from running after a failure.

Here's the exact command used to run the MinIO tests for example:
`go test . -timeout 30m -ginkgo.failFast --ginkgo.trace --ginkgo.progress --ginkgo.slowSpecThreshold=900 --ginkgo.v --ginkgo.focus="Medusa on MinIO" -v -p 1`

Reply:
I have not used `-ginkgo.failFast` before. That definitely helps. I wonder if that applies across the whole test suite or just within the current container. I hope the latter. I'll check that out.

Reply:
>If I can't get the cassandradatacenter resource to be ready, then there's no point in trying to access Cassandra or run a repair.

Absolutely. Once the CassandraDatacenter is ready though, we can perform multiple tests/verifications independently and possibly in parallel. For example, the C* cluster should be automatically registered in Reaper. If that check fails, it should not prevent us from checking that Stargate is deployed.

Reply:
I just ran the following using `-ginkgo.failFast`:

```go
var _ = FDescribe("A fake test", func() {
	Context("running multiple tests in context 1", func() {
		It("test 1", func() {
			Expect(1).To(Equal(11))
		})

		It("test 2", func() {
			Expect(1).To(Equal(2))
		})
	})

	Context("running multiple tests in context 2", func() {
		It("test 3", func() {
			Expect(1).To(Equal(3))
		})

		It("test 4", func() {
			Expect(1).To(Equal(3))
		})
	})
})
```

Here is the output:

```
• Failure [0.000 seconds]
A fake test
/Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/fake_test.go:8
  running multiple tests in context 1
  /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/fake_test.go:9
    test 1 [It]
    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/fake_test.go:10

    Expected
        <int>: 1
    to equal
        <int>: 11

    /Users/jsanda/Development/k8ssandra/k8ssandra/tests/unit/fake_test.go:11
------------------------------
```
It appears to apply across the entire test suite vs only the parent container.